From fd6e5bbb241720715cee737f534496d7c0ae9022 Mon Sep 17 00:00:00 2001
From: Roel Kluin <roel.kluin@gmail.com>
Date: Tue, 10 Aug 2010 14:29:19 -0700
Subject: USB: serial: io_ti.c: don't return 0 if writing the download
 record failed
Patch-mainline: v2.6.36
Git-commit: fd6e5bbb241720715cee737f534496d7c0ae9022

If the write download record failed we shouldn't return 0.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/io_ti.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/io_ti.c
+++ b/drivers/usb/serial/io_ti.c
@@ -1284,7 +1284,7 @@ static int download_fw(struct edgeport_s
 				kfree(header);
 				kfree(rom_desc);
 				kfree(ti_manuf_desc);
-				return status;
+				return -EINVAL;
 			}
 
 			/* Update I2C with type 0xf2 record with correct
